core/src/main/java/org/apache/accumulo/core/util/Merge.java

```diff
@@ -149,17 +148,37 @@ public void mergomatic(AccumuloClient client, String table, Text start, Text end
         throw new IllegalArgumentException("cannot merge tablets on the metadata table");
       }
       List<Size> sizes = new ArrayList<>();
-      long totalSize = 0;
-      // Merge any until you get larger than the goal size, and then merge one less tablet
-      Iterator<Size> sizeIterator = getSizeIterator(client, table, start, end);
-      while (sizeIterator.hasNext()) {
-        Size next = sizeIterator.next();
-        totalSize += next.size;
-        sizes.add(next);
-        if (totalSize > goalSize) {
-          totalSize = mergeMany(client, table, sizes, goalSize, force, false);
-        }
+      AtomicLong totalSize = new AtomicLong();
+
+      TableId tableId;
+      ClientContext context = (ClientContext) client;
+      try {
+        tableId = context.getTableId(table);
+      } catch (Exception e) {
+        throw new MergeException(e);
+      }
+      try (TabletsMetadata tablets = TabletsMetadata.builder(context).scanMetadataTable()
+          .overRange(new KeyExtent(tableId, end, start).toMetaRange()).fetch(FILES, PREV_ROW)
+          .build()) {
+        tablets.stream().map(tm -> {
+          long size = tm.getFilesMap().values().stream().mapToLong(DataFileValue::getSize).sum();
+          return new Size(tm.getExtent(), size);
+        }).forEach(next -> {
```

Comment:
This is a style comment so feel free to ignore it.  I would call `.iterator()` on the stream and use the while loop the old code had over the iterator in order to avoid having to use the AtomicLong to sum data in the lambda.

Reply:
I moved back to an iterator in a2122ee